### PR TITLE
Adjust `SHOW` statement

### PR DESCRIPTION
**Pull request by totalo**
For #14110.

Changes proposed in this pull request:
- adjust `SHOW` statement
- add more test case

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/dal/MySQLShowFunctionCodeStatement.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.apache.shardingsphere.sql.parser.sql.common.statement.dal.DALStatement;
+import org.apache.shardingsphere.sql.parser.sql.dialect.statement.mysql.MySQLStatement;
+
+/**
+ * MySQL show function code statement.
+ */
+@Getter
+@Setter
+@ToString
+public final class MySQLShowFunctionCodeStatement extends AbstractSQLStatement implements DALStatement, MySQLStatement {
```
Comment (taojintianxia) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/dal/MySQLShowFunctionCodeStatement.java:33]: why the other class name is plural, but not for this one

Comment (totalo) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/dal/MySQLShowFunctionCodeStatement.java:33]: The class name depends on the specific statement form, according to this ： https://dev.mysql.com/doc/refman/8.0/en/show-function-code.html

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/DALStatement.g4 [hunk 2]
```diff
@@ -33,31 +33,7 @@ explain
     | explainType? (explainableStatement | FOR CONNECTION connectionId)
     | ANALYZE (FORMAT EQ_ TREE)? select)
     ;
-
-showDatabases
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/DALStatement.g4:37]: @totalo Why remove these parse rule?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/BaseRule.g4 [hunk 3]
```diff
@@ -667,13 +667,15 @@ wrapperName
     ;
     
 functionName
-    : identifier
-    | (owner DOT_)? identifier
+    : (owner DOT_)? identifier
     ;
-    
+
+procName
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/BaseRule.g4:673]: What statement corresponds to the procName parsing rule?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/BaseRule.g4 (hunk 3 above)
Comment (totalo) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/BaseRule.g4:673]: `SHOW CREATE PROCEDURE procName`

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/DALStatement.g4 (hunk 2 above)
Comment (totalo) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/DALStatement.g4:37]: It is not deleted, but rearranged in the order of the official documents to facilitate subsequent proofreading.

Comment (strongduanmu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/DALStatement.g4:37]: Ok, I got it.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/BaseRule.g4 (hunk 3 above)
Comment (strongduanmu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/BaseRule.g4:673]: Do you think `procedureName` is better?

Comment (totalo) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/BaseRule.g4:673]: Yes!

**Review by strongduanmu — APPROVED** [review]
@totalo Good job.

**Pull request merged**